discovery-provider/src/api/v1/challenges.py

```diff
@@ -0,0 +1,62 @@
+import logging
+from flask_restx import Resource, Namespace, fields, reqparse
+from src.api.v1.models.common import favorite
+from src.api.v1.models.challenges import undisbursed_challenge
+
+from src.api.v1.helpers import (
+    decode_with_abort,
+    get_current_user_id,
+    make_response,
+    success_response,
+    extend_challenge_response,
+)
+
+
+from src.utils.redis_cache import cache
+from src.queries.get_undisbursed_challenges import get_undisbursed_challenges
+from src.utils.db_session import get_db_read_replica
+
+logger = logging.getLogger(__name__)
+
+ns = Namespace("challenges", description="Challenge related operations")
+
+get_undisbursed_challenges_route_parser = reqparse.RequestParser()
+get_undisbursed_challenges_route_parser.add_argument("limit", required=False, type=int)
```

Comment:
thoughts about if having offset would be useful as well?

Reply:
Yeah, we can add it, but I think cursor (keyset) base pagination might be more resilient? 
For reference, this currently uses `completed_blocknumber` as the pseudo offset, but maybe we should add another field for offset - will do. 
The things I'm worried about is if we add an offset and the challenges are being disbursed as we go, we could potentially skip challenges b/c of the offset.

Reply:
totally missed completed_blocknumber. makes sense